src/utils/messageManager.js

```diff
@@ -537,16 +589,22 @@ const manageRevealSecret = (msg, payment, messageKey) => {
     }
   } else if (msg.message_order === 7) {
     const { keccak256 } = ethers.utils;
-    const hasSameSecretHash =
-      keccak256(msg[messageKey].secret) === payment.secret_hash;
+    const { secret } = msg[messageKey];
+    const hasSameSecretHash = keccak256(secret) === payment.secret_hash;
     if (!hasSameSecretHash) return console.warn("Secret does not match");
     store.dispatch(
       addPendingPaymentMessage(msg.payment_id, msg.message_order, {
         message: msg[messageKey],
         message_order: msg.message_order,
       })
     );
-    store.dispatch(setPaymentSecret(payment.paymentId, msg[messageKey].secret));
+    store.dispatch(setPaymentSecret(payment.paymentId, secret));
+    const paymentAux = getPaymentByIdAndState(PENDING_PAYMENT, msg.payment_id);
+    const dataToPack = paymentAux.messages[1].message;
```

Comment:
We are 100% sure that paymentAux.messages has something at this point? this could cause an OutOfBounds error or something like that. Also it could be undefined, we should check at least that, unless this will never happend, or it shouldn't

Reply:
This property will never be null at this point.

Since the locked transfer (1) is always present with the payment, it is guaranteed that the message exists.
Also there is a check for the payment before processing some messages, that check that we have the payment beforehand.
Also messages is an Object, so it is not a bounds issue, it would be a null pointer.

So do not worry, it will not fail